stac_validator/validate.py

```diff
@@ -155,14 +159,20 @@ def extensions_validator(self, stac_type: str) -> dict:
     def custom_validator(self):
         # in case the path to custom json schema is local
         # it may contain relative references
-        schema = fetch_and_parse_schema(self.custom)
+
+        # deal with relative path in schema
         if os.path.exists(self.custom):
+            schema = fetch_and_parse_schema(self.custom)
             custom_abspath = os.path.abspath(self.custom)
             custom_dir = os.path.dirname(custom_abspath).replace("\\", "/")
             custom_uri = f"file:///{custom_dir}/"
             resolver = RefResolver(custom_uri, self.custom)
             jsonschema.validate(self.stac_content, schema, resolver=resolver)
         else:
+            if self.custom.startswith(".."):
+                file_directory = os.path.dirname(os.path.abspath(self.stac_file))
+                self.custom = os.path.join(file_directory, self.custom)
+                self.custom = os.path.abspath(os.path.realpath(self.custom))
             schema = fetch_and_parse_schema(self.custom)
             jsonschema.validate(self.stac_content, schema)
```

Comment:
I haven't dug into the logic here too deeply, but before I do, I have a higher-level suggestion: make the href absolute first, then do any work. E.g.:

```python
href = make_absolute_href(self.custom, self.stac_file)
# do the validation
```

You could use the functions from pystac, e.g. https://github.com/stac-utils/pystac/blob/25cdf9d00ccef0f3b4f6805258687936ef260ef1/pystac/utils.py#L249-L284. Since you don't depend on pystac, you could just copy the code to the utility module in this package and give credit back to pystac for the function?

I don't fully know if `RefResolver` would be required. IMO with this many open questions, it probably makes sense to add some more test cases for strange setups.